dockerworker/tarstream.go

```diff
@@ -0,0 +1,67 @@
+// +build docker
+
+package dockerworker
+
+import (
+	"archive/tar"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// a tarRewriter is a function that given a tar.Header and io.Reader returns a
+// tar.Header and io.Reader to be written in place of the entry given.
+//
+// If a nil header is returned  the entry is skipped, if no transformation is
+// desired the header and reader given can be returned with any changes.
+type tarRewriter func(header *tar.Header, r io.Reader) (*tar.Header, io.Reader, error)
```

Comment:
This sounds like a general purpose function not specific to the docker engine - can it go in its own package?

Can you provide a bit of context for why this is useful?

Reply:
When we download a tar docker image, we need to rewrite the manifest.json file before calling `docker load`, and then tar it again.